File: Promasy.Web.Frontend/src/services/api/cpv.ts
import { ErrorApiResponse, Fetch, Response } from "@/utils/fetch-utils";
import { buildQueryParameters } from "@/utils/url-params-utils";

export default {
  getList(
    parentId?: number,
    id?: number,
    searchText?: string,
  ): Promise<Response<Cpv[], ErrorApiResponse>> {
    return Fetch.Get<Cpv[], ErrorApiResponse>(
      `/api/cpv${ buildQueryParameters([
        [ "id", id?.toString() ],
        [ "parentId", parentId?.toString() ],
        [ "search", searchText ],
      ]) }`,
    );
  },
};

export interface Cpv {
  id: number;
  code: string;
  descriptionEnglish: string;
  descriptionUkrainian: string;
  level: number;
  isTerminal: boolean;
  parentId?: number;
}

File: Promasy.Domain/Vocabulary/Cpv.cs
namespace Promasy.Domain.Vocabulary;

public class Cpv
{
    public int Id { get; set; }
    public string Code { get; set; }
    public string DescriptionEnglish { get; set; }
    public string DescriptionUkrainian { get; set; }
    public int Level { get; set; }
    public bool IsTerminal { get; set; }

    public int? ParentId { get; set; }
    public Cpv Parent { get; set; }
}